### PR TITLE
Use open-wc tools

### PR DESCRIPTION
**Pull request by dlockhart**
This PR uses the [open-wc](https://open-wc.org/)'s [es-dev-server](https://open-wc.org/developing/es-dev-server.html#getting-started) which replaces `polymer serve` and their [testing tools](https://open-wc.org/testing/testing-karma.html#getting-started), which will replace `polymer test`.

I only converted one test file over just to give you an idea of what things will be like -- much simpler IMO. The open-wc Karma tests are set up to run alongside the old Polymer ones for now, and then once we convert everything over we can remove the Polymer stuff for good.

You'll notice that the open-wc tools come with their own Chai-based a11y testing (which uses aXe), so we'll be able to simplify our stuff once everything's converted.

### CODE REVIEW

File: mixins/localize-mixin.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 import {getDocumentLocaleSettings} from '@brightspace-ui/intl/lib/common.js';
-import IntlMessageFormat from 'intl-messageformat/src/main.js';
-window.IntlMessageFormat = IntlMessageFormat;
+import IntlMessageFormat from 'intl-messageformat';
```
Comment (dlockhart) [inline mixins/localize-mixin.js]: This shouldn't be in here, will fix.

File: README.md [hunk 2]
```diff
@@ -70,18 +70,12 @@ After cloning the repo, run `npm install` to install dependencies.
...
-If you don't have it already, install the [Polymer CLI](https://www.polymer-project.org/3.0/docs/tools/polymer-cli) globally:
-
-```shell
-npm install -g polymer-cli
-```
-
 ### Running the demos
 
-To start a [local web server](https://www.polymer-project.org/3.0/docs/tools/polymer-cli-commands#serve) that hosts the demo page and tests:
+To start a local web server that hosts the demo page and tests:
```
Comment (Scy-D2L) [inline README.md]: Would it still be helpful to have [a link to the server](https://open-wc.org/developing/es-dev-server.html#startserver) in here?

File: components/button/test/button-icon.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import '../button-icon.js';
+import { expect, fixture, oneEvent } from '@open-wc/testing';
+
+const basicFixture = '<d2l-button-icon icon="d2l-tier1:gear" text="Icon Button"></d2l-button-icon>';
+const disabledFixture = '<d2l-button-icon disabled icon="d2l-tier1:gear" text="Icon Button"></d2l-button-icon>';
+
+describe('d2l-button-icon', () => {
+
+	it('dispatches click event when clicked', async() => {
+		const el = await fixture(basicFixture);
```
Comment (svanherk) [inline components/button/test/button-icon.test.js]: Does this handle the waiting for `updateComplete` for us?  That's super nice

Comment (dlockhart) [reply components/button/test/button-icon.test.js]: It does indeed! Although I need to find a way to handle components using the localize mixin, as they need to wait a bit longer.

File: README.md (hunk 2 above)
Comment (dlockhart) [reply README.md]: Good call, done.

File: karma.conf.sauce.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	safari: {
+		base: 'SauceLabs',
+		browserName: 'safari',
+		platform: 'OS X 10.13'
+	}
+};
+
+module.exports = config => {
+	config.set(
+		merge(createDefaultConfig(config), {
```
Comment (dbatiste) [inline karma.conf.sauce.js]: Is it possible to use `Object.assign()`?  I think nodejs has supported it for a while.

Comment (dlockhart) [reply karma.conf.sauce.js]: I can give it a try. This is just following [their recommendation](https://open-wc.org/testing/testing-karma.html#manual).

File: components/button/test/button-icon.test.js (hunk 3 above)
Comment (dbatiste) [reply components/button/test/button-icon.test.js]: Could wait for a localize event.  For composite/app components, they would not know how many events to wait for though.  The async container mixin pattern comes to mind, though it was not written with localization specifically in mind, though it is applicable.

File: karma.conf.sauce.js (hunk 4 above)
Comment (dbatiste) [reply karma.conf.sauce.js]: Cool... yeah I think it would be nice to avoid if possible.

Comment (dlockhart) [reply karma.conf.sauce.js]: I don't think they're equivalent... deepmerge does a nested merge vs. `Object.assing()` only goes one level deep.

Comment (dbatiste) [reply karma.conf.sauce.js]: Ah 😞 .  I think those open-wc guys are pretty sharp, I was worried about that.

File: karma.conf.sauce.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+		platform: 'OS X 10.13'
+	},
+	safari: {
+		base: 'SauceLabs',
+		browserName: 'safari',
+		platform: 'OS X 10.13'
+	},
+	edge: {
+		base: 'SauceLabs',
+		browserName: 'microsoftedge',
```
Comment (dlockhart) [inline karma.conf.sauce.js]: Oh I was able to enable old-Edge testing for the Karma stuff! 🎉

**Pull request merged**